llvm/test/Feature/Repo/repo_prunning_dependents_3.ll

```diff
@@ -0,0 +1,28 @@
+; This test makes sure there is no redundant dependent GOs with the private linkage type in a compilation file.
+;
+; RUN: rm -f %t.db
+; RUN: env REPOFILE=%t.db llc -filetype=obj %S/Inputs/repo_pruning_dependent_with_private_linkage.ll -o %t
+; RUN: env REPOFILE=%t.db llc -filetype=obj %s -o %t1
+; RUN: env REPOFILE=%t.db pstore-dump -all-compilations %t.db | FileCheck %s
+
+target triple = "x86_64-pc-linux-gnu-repo"
+
+@str = private unnamed_addr constant [6 x i8] c"Ipsum\00", align 1
+
+define i32 @main() local_unnamed_addr !repo_ticket !0 {
+entry:
+  %puts.i = tail call i32 @puts(i8* getelementptr inbounds ([6 x i8], [6 x i8]* @str, i64 0, i64 0))
+  ret i32 0
+}
+
+declare i32 @puts(i8* nocapture readonly) local_unnamed_addr
+
+!repo.tickets = !{!0, !1, !2}
+
+!0 = !TicketNode(name: "main", digest: [16 x i8] c"G\A3\F3c\9D\10\142\9D\9F\89l\BCG\1DV", linkage: external, pruned: false)
+!1 = !TicketNode(name: "a", digest: [16 x i8] c"r\D5\E2\FD\BC\F9\97\AB\1E\03\CFr\C8\AB\A3\14", linkage: external, pruned: true)
+!2 = !TicketNode(name: "str", digest: [16 x i8] c"\A7\F3w\15S\7FP|b\09\FE\FCsg\9F\01", linkage: private, pruned: true)
+
+; CHECK:     name    : str
+; CHECK:     name    : str
+; CHECK-NOT: name    : str
```

Comment:
I'm not sure I understand this test. We compile twice and then dump all the compilations from the repo. Do check CHECK/CHECK-NOT statements ensure that "name:str" appears just twice? If so, that might be worthy of a comment because it wasn't obvious (to me).

Reply:
> 
> 
> I'm not sure I understand this test. We compile twice and then dump all the compilations from the repo. Do check CHECK/CHECK-NOT statements ensure that "name:str" appears just twice?
>

Yes, you are right.

> If so, that might be worthy of a comment because it wasn't obvious (to me).

Fixed, a comment has been added into the test.